File: 2270-number-of-ways-to-split-array/2270-number-of-ways-to-split-array.cpp
class Solution {
public:
    int waysToSplitArray(vector<int>& nums) {
        long sum = accumulate(nums.begin(), nums.end(), 0L);
        int ans = 0;
        int n = nums.size();
        long currSum = nums[0];
        for(int i = 1; i < n; i++) {
            if(currSum + currSum >= sum)
                ans++;
            currSum += nums[i];
        }
        return ans;
    }
};

File: 2270-number-of-ways-to-split-array/2270-number-of-ways-to-split-array.go
func waysToSplitArray(nums []int) int {
    var sum int64 = 0
    for _, num := range(nums) {
        sum += int64(num)
    }
    n := len(nums)
    ans := 0
    var currSum int64 = int64(nums[0])
    for i := 1; i < n; i++ {
        if(currSum >= sum - currSum) {
            ans++
        }
        currSum += int64(nums[i]);
    }
    return ans;
}